.github/workflows/qa-testbench.yaml

```diff
@@ -24,12 +35,17 @@ jobs:
     timeout-minutes: 45
     env:
       IMAGE: "gcr.io/zeebe-io/zeebe"
-      GENERATION_TEMPLATE: "${{ github.event.inputs.generation }}"
-      BRANCH_NAME: "${{ github.event.inputs.branch }}"
+      GENERATION_TEMPLATE: "${{ inputs.generation || 'Zeebe Snapshot' }}"
+      BRANCH_NAME: "${{ inputs.branch || github.ref_name || 'main' }}"
       ZEEBE_AUTHORIZATION_SERVER_URL: 'https://login.cloud.ultrawombat.com/oauth/token'
       ZEEBE_CLIENT_ID: 'S7GNoVCE6J-8L~OdFiI59kWM19P.wvKo'
 
     steps:
+      # We need to check out the evaluated branch and setup java (incl. maven), so we can retrieve the current project version
```

Comment:
:question: We we moved it, just for clarification

Reply:
Moved it to ensure that the checkout is the first thing we do, since it allows me to reuse the BRANCH_NAME env var (which has the resolved value) before it's modified. Since branch could be empty when the workflow is started (so that it defaults to the branch under test), we resolve it in the env var. It could probably be simplified, but I'd leave it for https://github.com/camunda/zeebe/issues/11021